src/shared/api/checkIpAccess.ts

```diff
@@ -0,0 +1,44 @@
+'use server'
+
+import { ENDPOINT } from '../model/constants'
+
+type Response = { denied?: 'true' | 'false' }
+
+const errorMsg = '[Error] IP 필터링 서비스를 사용할 수 없습니다.'
+const REVALIDATE_MS = 300 // 5분(300초)
+
+const checkIpAccess = async (clientIP: string) => {
+  const response = await fetch(
+    `${process.env.NEXT_PUBLIC_BASE_URL}${ENDPOINT.FILTER_IP}`,
+    {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ clientIP }),
+      cache: 'force-cache',
+      next: {
+        revalidate: REVALIDATE_MS,
+      },
+    },
+  )
+
+  if (!response.ok) {
+    throw new Error(errorMsg)
+  }
+
+  const data: Response = await response.json()
+
+  return isIpAccessDenied(data)
+}
+
+const isIpAccessDenied = (data: Response) => {
+  // 응답값이 false인 경우 차단되어야 함
+  const DENIED = false
+
+  if (!data.denied) {
```

Comment:
```suggestion
  if (data.denied === undefined) {
```

[P2] 혹시 undefined를 잡기 위한 로직이었다면 위와 같이 좀 더 명확하게 수정해보는 건 어떨까요?
37 라인과 41 라인에서 둘 다 `false` 값을 다루고 있다보니 Response 타입을 이해하기 전엔 로직을 이해하는 데에 어려움이 있었습니다...!

Reply:
@bicochan 
반영했습니다! 
https://github.com/frontend-opensource-project/speller-front/pull/101/commits/c8e1e40c312158bb17f12d0290fce2a2885cba1f
수정된 코드로 다시 한번 봐주시면 감사하겠습니다 😄